.github/workflows/action_test.yaml

```diff
@@ -0,0 +1,118 @@
+            payload: |
+              {
+                "inputs": {
+                    "payload": "{ trunkPath: \"./local-action/action-tests/trunk_shim.sh\", targetRefName: \"136/merge/\", targetCheckoutRef: \"refs/pull/136/merge/\", checkMode: \"pull_request\", trunkToken: \"fake-trunk-token\", pullRequest: { number: 136, head: { sha: \"69b531ac8f611e0ae73639ec606fbc23e8ead576\", ref: \"action-test-pr\" }, base: { sha: \"50039e906e0e53ce03b269e5e9e00879f4c6f05c\", ref: \"action-test\" } }, checkRunId: 14235603498, version: \"0.0.0\", githubToken:\"${{ github.token }}\", cache: true, cacheKey: \"trunk-check\", cachePath: \"~/.cache/trunk/tools/ruby\", trunkApiAddress: \"localhost-will-not-work:5021\", checkJobName: \"action-test/trunk-action/pull_request\", checkJobRunsOn: [\"self-hosted\", \"linux\", \"r6id2xlarge-org\"], checkWorkflowRunName: \"[action-test] Trunk Check Pull Request Mode\", targetCheckout: \"trunk-io/trunk-action\", concurrencyGroup: \"trunk-action/pull_request/136\", debug: true }"
+                  }
+              }
+            # payload: ${{ needs.prepare-matrix.outputs.payload-all }}
+          - description: all-action-test
+            payload: |
+              {
+                "inputs": {
+                    "payload": "{ trunkPath: \"./local-action/action-tests/trunk_shim.sh\", targetRefName: \"136/merge/\", targetCheckoutRef: \"refs/pull/136/merge/\", checkMode: \"pull_request\", trunkToken: \"fake-trunk-token\", pullRequest: { number: 136, head: { sha: \"69b531ac8f611e0ae73639ec606fbc23e8ead576\", ref: \"action-test-pr\" }, base: { sha: \"50039e906e0e53ce03b269e5e9e00879f4c6f05c\", ref: \"action-test\" } }, checkRunId: 14235603498, version: \"0.0.0\", githubToken:\"${{ github.token }}\", cache: true, cacheKey: \"trunk-check\", cachePath: \"~/.cache/trunk/tools/ruby\", trunkApiAddress: \"localhost-will-not-work:5021\", checkJobName: \"action-test/trunk-action/pull_request\", checkJobRunsOn: [\"self-hosted\", \"linux\", \"r6id2xlarge-org\"], checkWorkflowRunName: \"[action-test] Trunk Check Pull Request Mode\", targetCheckout: \"trunk-io/trunk-action\", concurrencyGroup: \"trunk-action/pull_request/136\", debug: true }"
+                  }
+              }
+            # payload: ${{ needs.prepare-matrix.outputs.payload-trunk-merge }}
+
+    steps:
+      - name: Checkout trunk-action
+        uses: actions/checkout@v3
+        with:
+          path: local-action
+
+      - name: Craft TEST_GITHUB_EVENT_PATH
+        shell: bash
+        # GITHUB_EVENT_PATH is the env var of path to file on runner that contains full webhook payload.
+        # we are replacing it with TEST_GITHUB_EVENT_PATH so we read our predefined payload for test cases.
+        run: |
+          TEST_GITHUB_EVENT_PATH=$(mktemp)
+          cat >> $TEST_GITHUB_EVENT_PATH <<EOF
+          ${{matrix.payload}}
+          EOF
+          echo "TEST_GITHUB_EVENT_PATH=TEST_GITHUB_EVENT_PATH" >>$GITHUB_ENV
```

Comment:
looking good! obvs this needs to become `echo TGEP=$TGEP >>env`

---

in addition, you could also switch to doing `TEST_GITHUB_EVENT_PATH=${GITHUB_ACTION_PATH}/action_tests/trunk_merge_test_payload.json`

---

to take it a step further, you can also do something like

```
TEST_GITHUB_EVENT_PATH=$(mktemp)
echo "{ \"inputs\": { \"payload\": $(cat ${{ matrix.payload }} | jq 'tojson') } }" >>$TEST_GITHUB_EVENT_PATH
echo TEST_GITHUB_EVENT_PATH=$TEST_GITHUB_EVENT_PATH >>$GITHUB_ENV
```

 and then all you would have to do would be to shove the contents of `inputs.payload` into the path in `matrix.payload`